ethereum/slogging.py

```diff
@@ -138,11 +85,228 @@ def pop_records(self):
 DEFAULT_LOGLEVEL = logging.INFO
 JSON_FORMAT = '%(message)s'
 PRINT_FORMAT = '%(levelname)s:%(name)s\t%(message)s'
+#FORMAT = "%(levelname)s:%(module)s %(asctime)-15s %(message)s"
+
+
+def get_configuration():
+    """
+    get a configuration (snapshot) that can be used to call configure
+    snapshot = get_configuration()
+    configure(**snapshot)
+    """
+    rootLogger = getLogger()
+    name_levels = [('', logging.getLevelName(rootLogger.level))]
+
+    for name, logger in list(rootLogger.manager.loggerDict.items()):
+        name_levels.append((name, logging.getLevelName(logger.level)))
+    config_string = ','.join('%s:%s' % x for x in name_levels)
+    #struct_root = get_logger().bind()
+    #log_json = bool([p for p in struct_root._processors if
+    #                 isinstance(p, structlog.processors.JSONRenderer)])
+    return dict(config_string=config_string, log_json=rootLogger.log_json)
+
+# helpers
+known_loggers = set()  # know to structlog (i.e. maybe not yet initialized w/ logging)
+
+
+def get_logger_names():
+    # logging.Logger.manager.loggerDict.keys() # used ones
+    # initialized at module load get_logger
+    return sorted(known_loggers, key=lambda x: '' if not x else x)
+
+#----------------------------- from standart logging module width some modifications ----------------------------
+class ethLogger(logging.Logger):
+    def __init__(self, name, level=DEFAULT_LOGLEVEL):
+        super(ethLogger, self).__init__(name, level=level)
+
+    def is_active(self, level_name='trace'):
+        return self.isEnabledFor(logging._checkLevel(level_name.upper()))
+        #getLogger()
+        #self.setLevel(getattr(logging, level_name.upper()))
 
 
+class RootLogger(ethLogger):
+    """
+    A root logger is not that different to any other logger, except that
+    it must have a logging level and there is only one instance of it in
+    the hierarchy.
+    """
+    def __init__(self, level, log_json=False):
+        """
+        Initialize the logger with the name "root".
+        """
+        ethLogger.__init__(self, "root", level)
+        self.log_json = False
```

Comment:
The parameter `log_json` is ignored. Either remove it or assign it to the instance variable.